File: Lab4/app/src/main/java/com/example/lab4/MainActivity2.java
package com.example.lab4;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioGroup;

public class MainActivity2 extends AppCompatActivity {

    private Button send_btn;
    private EditText set_drink;
    private RadioGroup rg1,rg2;

    private String sugar="無糖";
    private String ice_opt="去冰";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        rg1= findViewById(R.id.radioGroup);
        rg1.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener(){
            @Override
            public void onCheckedChanged(RadioGroup group, int i) {
                switch(i){
                    case R.id.radioButton5:
                        sugar="無糖";
                        break;
                    case R.id.radioButton6:
                        sugar="少糖";
                        break;
                    case R.id.radioButton7:
                        sugar="半糖";
                        break;
                    case R.id.radioButton8:
                        sugar="全糖";
                        break;
                }
            }

        });

        rg2=findViewById(R.id.radioGroup2);
        rg2.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int i) {
                switch(i){
                    case R.id.radioButton9:
                        ice_opt="去冰";
                        break;
                    case R.id.radioButton10:
                        ice_opt="微冰";
                        break;
                    case R.id.radioButton11:
                        ice_opt="少冰";
                        break;
                    case R.id.radioButton12:
                        ice_opt="正常冰";
                        break;
                }
            }
        });

        send_btn=findViewById(R.id.btn_send);
        send_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                set_drink= findViewById(R.id.ed_drink);
                String drink = set_drink.getText().toString();
                Intent i = new Intent();
                Bundle b = new Bundle();
                b.putString("sugar",sugar);
                b.putString("drink",drink);
                b.putString("ice",ice_opt);
                i.putExtras(b);
                setResult(101,i);
                finish();
            }
        });
    }
}

File: Lab4_withKotlin/app/src/main/java/com/stella/lab4_withkotlin/MainActivity2.kt
package com.stella.lab4_withkotlin

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity

class MainActivity2 : AppCompatActivity() {

    lateinit var send_btn: Button
    lateinit var set_drink: EditText
    lateinit var rg1: RadioGroup
    lateinit var rg2: RadioGroup

    private var sugar = "無糖"
    private var ice_opt = "去冰"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        rg1 = findViewById(R.id.radioGroup)
        rg1.setOnCheckedChangeListener(RadioGroup.OnCheckedChangeListener { group, i ->
            when (i) {
                R.id.radioButton5 -> sugar = "無糖"
                R.id.radioButton6 -> sugar = "少糖"
                R.id.radioButton7 -> sugar = "半糖"
                R.id.radioButton8 -> sugar = "全糖"
            }
        })
        rg2 = findViewById<RadioGroup>(R.id.radioGroup2)
        rg2.setOnCheckedChangeListener(RadioGroup.OnCheckedChangeListener { group, i ->
            when (i) {
                R.id.radioButton9 -> ice_opt = "去冰"
                R.id.radioButton10 -> ice_opt = "微冰"
                R.id.radioButton11 -> ice_opt = "少冰"
                R.id.radioButton12 -> ice_opt = "正常冰"
            }
        })
        send_btn = findViewById(R.id.btn_send)
        send_btn.setOnClickListener(View.OnClickListener {
            set_drink = findViewById(R.id.ed_drink)
            val drink = set_drink.getText().toString()
            val i = Intent()
            val b = Bundle()
            b.putString("sugar", sugar)
            b.putString("drink", drink)
            b.putString("ice", ice_opt)
            i.putExtras(b)
            setResult(101, i)
            finish()
        })
    }
}